src/Blorc.OpenIdConnect.Tests/Services/ObjectExtensionsFacts.cs

```diff
@@ -0,0 +1,130 @@
+﻿namespace Blorc.OpenIdConnect.Tests.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Claims;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class ObjectExtensionsFacts
+    {
+        public class The_AsClaims_Method
+        {
+            [Test]
+            public void Collects_Roles_AsClaims()
+            {
+                var user = new User<Profile>
+                {
+                    AccessToken = "1234567890",
+                    Profile = new Profile
+                    {
+                        Roles = new[] { "Administrator", "System Administrator" },
+                        Email = "jane.doe@blorc.com",
+                        EmailVerified = true,
+                        FamilyName = "Doe",
+                        GivenName = "Jane",
+                        Name = "Jane Doe",
+                        PreferredUsername = "jane.doe"
+                    },
+                    ExpiresAt = 10,
+                    SessionState = "alskjdhflaskjdhflaksjdhqwpoyir",
+                    TokenType = "Bearer"
+                };
+
+                var claims = user.AsClaims().Where(claim => claim.Type == ClaimTypes.Role).ToList();
+
+                Assert.IsNotEmpty(claims);
+                foreach (var claim in claims)
```

Comment:
I think we should do this the other way. In order to test whether the claims are returning the correct data, we should loop user.Profile.Roles (we know these values are correct) and make sure the claims are available.

Reply:
Yes I understand your point